### ISSUE TITLE
bug: 

### ISSUE THREAD

**Issue by muaviaUsmani** [ionic-team/ionic#20669]
# Bug Report

**Ionic version:**

[x] **5.0.4**

**Current behavior:**
Declaring a reference to tabs elements as `HTMLIonTabsElement` and then using it on an `IonTabs` element gives an error saying 
```
Type 'RefObject<HTMLIonTabsElement>' is not assignable to type 'RefObject<IonTabs>'.
```

**Expected behavior:**
Should be able to create a ref using `HTMLIonTabsElement` type and assign it to a Tabs element.

**Steps to reproduce:**
1. Create a ref variable with type `HTMLIonTabsElement`.
2. Assign it to the Tabs element.

**Related code:**

```typescript
let tabsRef = useRef<HTMLIonTabsElement>(null);

return (
    <IonTabs ref={tabsRef}>
    </IonTabs>
);
```

**Other information:**

**Ionic info:**

```
Ionic:

   Ionic CLI       : 6.1.0 (/Users/abc/.nvm/versions/node/v10.16.3/lib/node_modules/@ionic/cli)
   Ionic Framework : @ionic/react 5.0.4

Capacitor:

   Capacitor CLI   : 1.4.0
   @capacitor/core : 1.4.0

Utility:

   cordova-res : not installed
   native-run  : not installed

System:

   NodeJS : v10.16.3 (/Users/abc/.nvm/versions/node/v10.16.3/bin/node)
   npm    : 6.13.0
   OS     : macOS Catalina

```

**Comment by aivinog1** [comment]
@liamdebeasi Hello, can I take this? For me this problem is clear.

**Comment by liamdebeasi** [comment]
Yes, thank you! For a guide on how to create a pull request and test this project locally to see your changes, see our [contributing documentation](https://ionicframework.com/docs/building/contributing).

Please let me know if you have any questions.

**Comment by aivinog1** [comment]
@liamdebeasi Thanks :)

**Comment by liamdebeasi** [comment]
Hey there @muaviaUsmani, 

What are you trying to accomplish with this use case? The `IonTab` in Ionic React is a custom React component, so it is not supposed to have `HTMLIonTabElement` as a type.

The PR for this (https://github.com/ionic-team/ionic/pull/21376) is not something we can currently merge because it takes out the custom React ion-tabs and uses the Ionic Core ion-tabs instead.

**Comment by muaviaUsmani** [comment]
The goal was to be able to programmatically change tabs using select method as mentioned in the documentation https://ionicframework.com/docs/api/tabs#methods.

The problem I would point out here is that the ability to use `HTMLIonTabElement` should be removed from documentation if it is not available, or not planned to be included.

**Comment by elylucas** [comment]
Hi @muaviaUsmani,

We have a `IonTabsContext` context you can use that fills in some of the missing methods from the IonTabs ref:

```
  const tabsContext = useContext(IonTabsContext); // IonTabsContext is imported from @ionic/react

  const changeTab = () => {
    tabsContext.selectTab('tab2');
  }
```

Let us know if that fixes your issue. And sorry, this isn't documented well either, we will fix that.

Thanks
